### PR TITLE
Sqrt series functions

### PR DESCRIPTION
**Pull request by wbhart**
* Add issquare, issquare_with_sqrt for Abs/RelSeries
* Add sqrt_classical_char2 for Abs/RelSeries
* Add issquare, issquare_with_sqrt for Float64
* Add test code for mod p series sqrt (incl. p = 2)

### CODE REVIEW

File: src/RelSeries.jl [hunk 1]
```diff
@@ -1043,32 +1043,72 @@ end
...
+      return sqrt_classical_char2(a, check=check)
+   end
    aval2 = div(aval, 2)
    prec = precision(a) - aval
    if prec == 0
       asqrt = parent(a)()
       asqrt = set_precision!(asqrt, aval2)
       asqrt = set_valuation!(asqrt, aval2)
-      return asqrt
+      return true, asqrt
```
Comment (tthsqe12) [inline src/RelSeries.jl:1100]: There are too many untested corner cases like this. Add some test cases to make sure that these bits of code are actually doing the correct thing. Code coverage actually decreases with the current state of this PR.

**Review by tthsqe12 — CHANGES_REQUESTED** [review]

File: src/julia/Float.jl [hunk 2]
```diff
@@ -120,6 +120,14 @@ function Base.sqrt(a::AbstractFloat; check::Bool=true)
...
 end
 
+function issquare(a::AbstractFloat)
+   return true
+end
+
+function issquare_with_sqrt(a::AbstractFloat)
+   return true, Base.sqrt(a)
+end
+
```
Comment (tthsqe12) [inline src/julia/Float.jl:130]: I guess the remaining corner cases we can fix later if there are issues, but these two is glaring. `Base.sqrt` does not think that `Float64(-1)` is a square.

**Review by tthsqe12 — CHANGES_REQUESTED** [review]

File: src/julia/Float.jl (hunk 2 above)
Comment (wbhart) [reply src/julia/Float.jl:130]: Thanks. Those are indeed bugs on my part!

I'm attempting to get the other code cov cases covered, but it's not trivial to do locally. I thought I had them covered, but well, you can see that's not the case. Probably some silly errors on my part.

Comment (wbhart) [reply src/julia/Float.jl:130]: Oh, I think I got them this time.

Comment (wbhart) [reply src/julia/Float.jl]: Hopefully it is all ok now. Time to do Laurent and Puiseux series, which I'll do on a separate PR.

**Review by tthsqe12 — APPROVED** [review]
